backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -1525,15 +1536,15 @@ export class ExperimentAssignmentService {
     }
   }
 
-  private assignExperiment(
+  private async assignExperiment(
```

Comment:
because we are going to sometimes potentially be calling out to another service, I had to make assignExperiment return a Promise, which causes me to have to modify this function in a bunch of places. I don't love this, but I wasn't sure how else to accomplish it. Ideas welcome.